packages/lit-element/src/hydrate-support.ts

```diff
@@ -30,31 +31,75 @@ interface PatchableLitElement extends HTMLElement {
 }: {
   LitElement: PatchableLitElement;
 }) => {
-  // Capture whether we need hydration or not
-  const createRenderRoot = LitElement.prototype.createRenderRoot;
-  LitElement.prototype.createRenderRoot = function (this: PatchableLitElement) {
+  const observedAttributes = Object.getOwnPropertyDescriptor(
+    Object.getPrototypeOf(LitElement),
+    'observedAttributes'
+  )!.get!;
+
+  // Add `defer-hydration` to observedAttributes
+  Object.defineProperty(LitElement, 'observedAttributes', {
+    get() {
+      return [...observedAttributes.call(this), 'defer-hydration'];
+    },
+  });
+
+  // Enable element when 'defer-hydration' attribute is removed
+  const attributeChangedCallback =
+    LitElement.prototype.attributeChangedCallback;
+  LitElement.prototype.attributeChangedCallback = function (
+    name: string,
+    old: string | null,
+    value: string | null
+  ) {
+    if (name === 'defer-hydration' && value === null) {
+      connectedCallback.call(this);
+    } else {
+      attributeChangedCallback.call(this, name, old, value);
+    }
+  };
+
+  // Override `connectedCallback` to capture whether we need hydration, and
+  // defer `enableUpdate()` if the 'defer-hydration' attribute is set
+  const connectedCallback = LitElement.prototype.connectedCallback;
+  LitElement.prototype.connectedCallback = function (
+    this: PatchableLitElement
+  ) {
     if (this.shadowRoot) {
       this._$needsHydration = true;
+    }
```

Comment:
Why does `_$needsHydration` need to be set here?

Reply:
Removed